src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -566,43 +569,36 @@ CreateFixPartitionShardIndexNamesTaskList(Oid parentRelationId, Oid partitionRel
 			WorkerFixPartitionShardIndexNamesCommandList(parentShardId,
 														 parentIndexIdList,
 														 partitionRelationId);
-
 		if (queryStringList != NIL)
 		{
 			Task *task = CitusMakeNode(Task);
 			task->jobId = INVALID_JOB_ID;
 			task->taskId = taskId++;
-
 			task->taskType = DDL_TASK;
 
-			/*
-			 * There could be O(#partitions * #indexes) queries in
-			 * the queryStringList.
-			 *
-			 * In order to avoid round-trips per query in queryStringList,
-			 * we join the string and send as a single command via the UDF.
-			 * Otherwise, the executor sends each command with one
-			 * round-trip.
-			 */
-			char *string = StringJoin(queryStringList, ';');
-			StringInfo commandToRun = makeStringInfo();
-
-			appendStringInfo(commandToRun,
-							 "SELECT pg_catalog.citus_run_local_command($$%s$$)", string);
-			SetTaskQueryString(task, commandToRun->data);
+			char *prefix = "SELECT pg_catalog.citus_run_local_command($$";
```

Comment:
interesting trick

Reply:
Given the command here could be really large, I observed non-negligible improvements by avoiding an extra copy of *almost* the same command